src/OversetSimulation.h

```diff
@@ -109,6 +111,30 @@ class OversetSimulation
         m_is_adaptive_holemap_alg = alg;
         if (m_is_adaptive_holemap_alg == true) m_tg.setHoleMapAlgorithm(1);
     }
+
+    void set_composite_num(int num_composite)
+    {
+        m_num_composite_bodies = num_composite;
+        m_tg.setNumCompositeBodies(num_composite);
+    }
+
+    void set_composite_body(
+        int body_index,
+        int num_body_tags,
+        std::vector<int> bodytags,
```

Comment:
passing in the vectors like this implies a copy I think. You can pass in `const .. & ...` Though you have to make sure that `registerCompositeBody` then doesn't assume that data lives and try to access it later (ie does it make a copy or does it just hold the pointer? and is that pointer alive for the rest of the sim)

Reply:
This function calls `tioga:registerCompositeBody`, which is a Tioga function that expects integer pointers (`int *`). So I pass Tioga a pointer to these vectors using the 'data()' function. The information that is passed is saved into the 'compositeBody' object in Tioga, and these pointers do not have to stay alive for the rest of execution. I do not think that I can pass a const vector reference to Tioga without modifying 'registerCompositeBody', unless I am mistaken.